pkg/listwatch/listwatch.go

```diff
@@ -195,14 +221,14 @@ func (mw *multiWatch) ResultChan() <-chan watch.Event {
 // It stops all of the underlying watch.Interfaces and closes the backing chan.
 // Can safely be called more than once.
 func (mw *multiWatch) Stop() {
-	mw.mu.Lock()
-	defer mw.mu.Unlock()
-	if !mw.stopped {
+	select {
+	case <-mw.stopped:
+		// nothing to do, we are already stopped
+	default:
 		for _, stop := range mw.stoppers {
 			stop()
```

Comment:
Strictly, if multiple goroutines invoke `Stop()` this could be invoked multiple times. I guess it's not worth guarding this, as I think all `Stop()` impls are reentrant, but nevertheless pointing this out here.